docs/src/man/basics.md

```diff
@@ -1510,31 +1510,69 @@ julia> german[Not(5), r"S"]
-  target column names is automatically generated.
-- `source_column => target_column_name` renames the `source_column` to `target_column_name`.
-- `source_column` just keep the source column as is in the result without any transformation;
-
-These rules are typically called transformation mini-language.
+- `transform`: creates a new data frame containing all source columns and columns
+ that result from the transformation;
+- `transform!`: adds columns that result from the transformation to an existing data frame;
+- `select`: creates a new data frame with only columns that result from the transformation;
+- `select!`: modifies an existing data frame in place, retaining only columns that result
+ from the transformation;
+- `combine`: the same as `select` but the number of rows may be reduced by the transformation;
+
+!!! Note
+    `combine` is the only transformation function to return a different number of rows
+     than the source data frame. The difference in behavior between `combine` and `select` 
+     will become clear in the following examples. There is no `combine!` function because ...
+
+All of the functions above use the same syntax which is commonly
+ `transform(source_dataframe, transformation)`.
+ The `transformation` argument is a `Pair` which defines the transformation to be applied
+  to the `source_dataframe`, and it can take any of the forms listed below:
+
+- `source_column_selector => function => new_column_name`:  
+passes source column(s) to function and names the
+resulting column(s) `new_column_name`;
+- `source_column_selector => function`:  
+passes source column(s) to function and automatically names the resulting column(s);
+- `source_column_selector => new_column_name`:  
+renames source column(s);
+- `source_column_selector`:  
+selects source column(s) without transforming them
+(used often with `select` for moving or deleting columns);
+
+The most basic `source_column_selector` is a column name, but there are many more ways to
+select columns as explained in the
+[Indexing API documentation](https://dataframes.juliadata.org/stable/lib/indexing/#Indexing).
+
+Here `function` is a function which operates on an entire data frame
+column passed as type `Vector`.
+If you instead want to apply a function to each element in the column,
+then you can wrap your element-wise function in `ByRow` like
+`function = ByRow(my_elementwise_function)`,
+which will conveniently collect your element-wise function results
+into a `Vector`.
+When multiple columns are selected,
+the `function` will receive the columns as multiple arguments in the
+order they are selected like `function(column1, column2, column3)`.
+Alternatively, the selected columns can be "slurped" into a
+single argument using `function(columns...)`.
+In more advanced usage, the `function` itself can use column selectors
+in its definition.
+
+`new_column_name` may be a `String` or a `Symbol`.
+(*Soon `new_column_name` will also accept a renaming function.*)
```

Comment:
Can you only use the renaming function in the `source_column_selector => function => new_column_names` form? If I use a renaming function in `source_column_selector => new_column_names`, it thinks I am trying to do `source_column_selector => function`.

Reply:
Can you please share the code you are using as I am not clear what is not working for you?

If in `a => b` the `b` part is a function it will be interpreted as transformation specification. It is the only way it could be interpreted, as otherwise how could DataFrames.jl know if `b` should be a transformation or a column renaming function?

If you pass `a => b => c` then `c` can be one of:
* `AsTable`
* list of column names
* a function taking input column names from `a` and producing target column names

Reply:
You confirmed my suspicion. I didn't know if there was a special way around the `a => b` renaming limitation.
```julia
julia> using DataFrames

julia> df = DataFrame(x=[1,7,3], y=[6,0,2])
3×2 DataFrame
 Row │ x      y     
     │ Int64  Int64 
─────┼──────────────
   1 │     1      6
   2 │     7      0
   3 │     3      2

julia> newname(x::String) = x*"_new"
newname (generic function with 1 method)

julia> transform(df, :x => (x -> 2x) => newname)
3×3 DataFrame
 Row │ x      y      x_new 
     │ Int64  Int64  Int64
─────┼─────────────────────
   1 │     1      6      2
   2 │     7      0     14
   3 │     3      2      6

julia> transform(df, :x => newname)
ERROR: MethodError: no method matching newname(::Vector{Int64})
Closest candidates are:
  newname(::String) at REPL[3]:1
Stacktrace:
 [1] _transformation_helper(df::DataFrame, col_idx::Int64, ::Base.RefValue{Any})
   @ DataFrames C:\Users\nboyer.AIP\.julia\dev\DataFrames\src\abstractdataframe\selection.jl:397
 [2] select_transform!(::Base.RefValue{Any}, df::DataFrame, newdf::DataFrame, transformed_cols::Set{Symbol}, copycols::Bool, allow_resizing_newdf::Base.RefValue{Bool})
   @ DataFrames C:\Users\nboyer.AIP\.julia\dev\DataFrames\src\abstractdataframe\selection.jl:575
 [3] _manipulate(df::DataFrame, normalized_cs::Vector{Any}, copycols::Bool, keeprows::Bool)
   @ DataFrames C:\Users\nboyer.AIP\.julia\dev\DataFrames\src\abstractdataframe\selection.jl:1444
 [4] manipulate(::DataFrame, ::Any, ::Vararg{Any, N} where N; copycols::Bool, keeprows::Bool, renamecols::Bool)
   @ DataFrames C:\Users\nboyer.AIP\.julia\dev\DataFrames\src\abstractdataframe\selection.jl:1372
 [5] #select#466
   @ C:\Users\nboyer.AIP\.julia\dev\DataFrames\src\abstractdataframe\selection.jl:992 [inlined]
 [6] transform(df::DataFrame, args::Any; copycols::Bool, renamecols::Bool)
   @ DataFrames C:\Users\nboyer.AIP\.julia\dev\DataFrames\src\abstractdataframe\selection.jl:1070
 [7] transform(df::DataFrame, args::Any)
   @ DataFrames C:\Users\nboyer.AIP\.julia\dev\DataFrames\src\abstractdataframe\selection.jl:1059
 [8] top-level scope
   @ REPL[6]:1
```

Reply:
> the `a => b` renaming limitation.

If you pass one column (as this is what you do) just write `a => identity => b` or just `a => newname(string(a))`.

The function interface is intended for the cases where you rather perform multiple transformations in one shot.